src/sentry/mediators/sentry_app_installations/authorizer.py

```diff
@@ -13,31 +14,53 @@
 
 TOKEN_LIFE_IN_HOURS = 8
 
+AUTHORIZATION = 'authorization_code'
+REFRESH = 'refresh_token'
+
+GRANT_TYPES = (
+    AUTHORIZATION,
+    REFRESH,
+)
+
 
 class Authorizer(Mediator):
     install = Param('sentry.models.SentryAppInstallation')
     grant_type = Param(six.string_types)
-    code = Param(six.string_types)
+    code = Param(six.string_types, required=False)
+    refresh_token = Param(six.string_types, required=False)
     client_id = Param(six.string_types)
     user = Param('sentry.models.User')
 
     def call(self):
+        self._validate_code_or_refresh_token()
         self._validate_grant_type()
         self._validate_install()
         self._validate_sentry_app()
-        self._validate_grant()
-        return self.exchange()
 
-    def exchange(self):
+        if self.grant_type == AUTHORIZATION:
+            self._validate_grant()
+
+        if self.grant_type == REFRESH:
+            self._validate_refresh_token()
+            self._expire_active_token()
+
+        return self.create_api_token()
+
+    def create_api_token(self):
         return ApiToken.objects.create(
             user=self.user,
             application=self.application,
             scope_list=self.sentry_app.scope_list,
-            expires_at=(datetime.now() + timedelta(hours=TOKEN_LIFE_IN_HOURS)),
+            expires_at=(datetime.utcnow() + timedelta(hours=TOKEN_LIFE_IN_HOURS)),
         )
 
+    def _validate_code_or_refresh_token(self):
```

Comment:
would it make sense to combine `_validate_code_or_refresh_token` and `_validate_grant_type` and do something like 
```python
is_refresh_token = self.refresh_token and self.grant_type == 'refresh_token' 
is_authorization_code = self.code and self.grant_type == 'authorization_code' 

if not is_refresh_token or is_authorization_code:
    raise APIUnauthorized
```

cause what happens if someone accidentally sends `code` and `refresh_token`